internal/controlplane/handlers_profile.go

```diff
@@ -273,41 +276,34 @@ func getProfilePBFromDB(
 	return nil, fmt.Errorf("profile not found")
 }
 
-func (s *Server) getRuleEvalEntityInfo(
-	ctx context.Context,
-	entityType *db.NullEntities,
-	selector *uuid.NullUUID,
+// TODO: We need to replace this with a more generic method that can be used for all entities
+// probably coming from the properties.
+func getRuleEvalEntityInfo(
 	rs db.ListRuleEvaluationsByProfileIdRow,
-	projectID uuid.UUID,
+	efp *entmodels.EntityForProperties,
 ) map[string]string {
-	l := zerolog.Ctx(ctx)
 	entityInfo := map[string]string{}
 
-	if rs.RepositoryID.Valid {
-		// this is always true now but might not be when we support entities not tied to a repo
-		entityInfo["repo_name"] = rs.RepoName.String
-		entityInfo["repo_owner"] = rs.RepoOwner.String
-		entityInfo["provider"] = rs.Provider.String
-		entityInfo["repository_id"] = rs.RepositoryID.UUID.String()
+	if owner := efp.Properties.GetProperty(ghprop.RepoPropertyOwner); owner != nil {
+		entityInfo["repo_owner"] = owner.GetString()
 	}
-
-	if !selector.Valid || !entityType.Valid {
-		return entityInfo
+	if name := efp.Properties.GetProperty(ghprop.RepoPropertyName); name != nil {
+		entityInfo["repo_name"] = name.GetString()
 	}
 
-	if entityType.Entities == db.EntitiesArtifact {
-		artifact, err := s.store.GetArtifactByID(ctx, db.GetArtifactByIDParams{
-			ID:        selector.UUID,
-			ProjectID: projectID,
-		})
-		if err != nil {
-			l.Err(err).Msg("error getting artifact by ID")
-			return entityInfo
-		}
-		entityInfo["artifact_id"] = artifact.ID.String()
-		entityInfo["artifact_name"] = artifact.ArtifactName
-		entityInfo["artifact_type"] = artifact.ArtifactType
-		entityInfo["provider"] = artifact.ProviderName
+	// entityInfo["artifact_id"] = artifact.ID.String()
+	// entityInfo["artifact_name"] = artifact.ArtifactName
+	// entityInfo["artifact_type"] = artifact.ArtifactType
+
+	entityInfo["provider"] = rs.Provider
+	entityInfo["entity_type"] = efp.Entity.Type.ToString()
+	entityInfo["entity_id"] = rs.EntityID.String()
+
+	// temporary: These will be replaced by entity_id
```

Comment:
as well as these could be simplified?

Reply:
I think we'll have a set of properties we output and have a general way of displaying them, so this will be replaced with a provider-specific call IMO.